File: 05_algebrogram/algebrogram.py
# algebrogram: ABCD*D = DCBA
# brute force solution
for a in range(10):
    for b in range(10):
        if b == a:
            continue
        for c in range(10):
            if c == b or c == a:
                continue
            for d in range(10):
                if d == c or d == b or d == a:
                    continue
                left = (a * 1000 + b * 100 + c * 10 + d) * d
                right = d * 1000 + c * 100 + b * 10 + a
                if left == right:
                    r = a, b, c, d
                    break
if r:
    print("abcd*d = dcba")
    print(
        str(r[0])
        + str(r[1])
        + str(r[2])
        + str(r[3])
        + "*"
        + str(r[3])
        + " = "
        + str(r[3])
        + str(r[2])
        + str(r[1])
        + str(r[0])
    )
else:
    print("none")

File: 05_algebrogram/algebrogram.php
<?php
/* 
 * algebrogram: ABCD*D = DCBA 
 * brute force solution
 */
for ($a = 0; $a <= 9; $a++) {
    for ($b = 0; $b <= 9; $b++) {
        if ($b == $a) {
            continue;
        }
        for ($c = 0; $c <= 9; $c++) {
            if ($c == $b || $c == $a) {
                continue;
            }
            for ($d = 0; $d <= 9; $d++) {
                if ($d == $c || $d == $b || $d == $a) {
                    continue;
                }
                if ((int)"$a$b$c$d" * (int)"$d" == (int)"$d$c$b$a") {
                    $r = "$a$b$c$d";
                    break;
                }
            }
        }
    }
}
if (!empty($r)) {
    echo "abcd*d = dcba\n";
    echo $r[0] . $r[1] . $r[2] . $r[3] . "*" . $r[3] . " = " . $r[3] . $r[2] . $r[1] . $r[0] . "\n";
} else {
    echo "none\n";
}
